seonjeong/app.js

```diff
@@ -51,6 +51,145 @@ app.post("/users", async (req, res) => {
   return res.status(201).json({ message: "successfully created" });
 });
 
+app.post("/posts", async (req, res) => {
+  const { title, content, imageUrl, userId } = req.body;
+
+  await appDataSource.query(
+    `INSERT INTO posts (
+      title,
+      content,
+      image_url,
+      user_id
+    ) VALUES (?, ?, ?, ?);`,
+    [title, content, imageUrl, userId]
+  );
+  return res.status(201).json({ message: "postCreated" });
+});
+//동기: 융통성x, 비동기: 융통성o
+//async + await
+
+app.get("/search", async (req, res) => {
+  const result = await appDataSource.query(
+    `SELECT
+    users.id as userId,
+    users.profile_image as userProfileImage,
+    posts.id as postingId,
+    posts.image_url as postingImageUrl,
+    posts.content as postingContent
+    FROM users INNER JOIN posts
+    ON users.id = posts.user_id
+    `
+  );
+
+  return res.status(200).json({ data: result });
+});
+
+app.get("users/userId/posts", async (req, res) => {
+  const { userId } = req.params;
+
+  const result = await appDataSource.query(
+    `
+  SELECT
+    u.id as userId,
+    u.profile_image as userProfileImage,
+    JSON_ARRAYAGG(
+      JSON_OBJECT(
+        "postingId", p.id,
+        "postingImageUrl", p.image_url,
+        "postingTitle", p.title,
+        "postingContent", p.content
+      )
+    ) as postings
+    FROM users as u
+    JOIN posts as p ON p.user_id = u.id
+    WHERE u.id = ?
+    GROUP BY u.id
+    `,
+    [userId]
+  );
+
+  return res.status(200).json({ data: result });
+});
+//ARRAYAGG: 객체를 배열로 묶어주는 역할
+//GROUP BY: 특정 컬럼을 그룹화함
+
+app.patch("/posts/postId", async (req, res) => {
+  const { postId } = req.params;
+  const { content } = req.body;
+
+  await appDataSource.query(
+    `
+    UPDATE posts
+    SET content = ?
+    WHERE id = ?
+  `,
+    [content, postId]
+  );
+
+  const result = await appDataSource.query(
+    `
+    SELECT
+      u.id as userId,
+      u.name as userName,
+      p.id as postingId,
+      p.title as postingTitle,
+      p.content as postingContent
+    FROM posts p
+    JOIN users u ON u.id = p.user_id
+    WHERE p.id = ?
+  `,
+    [postId]
+  );
+
+  return res.status(201).json({ message: "postModify" });
```

Comment:
1. 수정이 완료 됐을 때 status code는 200을 사용해 주세요.
2. 수정 후에 select를 통해 수정된 결과를 가져왔는데, 단순히 message만 주고 있습니다. 수정된 결과에 대한 데이터도 같이 보내주면 좋을 것 같습니다.
```suggestion
  return res.status(200).json({ message: "postModify", data: result});
```